fate/src/main/java/org/apache/accumulo/fate/AdminUtil.java

```diff
@@ -249,21 +310,36 @@ public FateStatus getStatus(ReadOnlyTStore<T> zs, IZooReader zk, String lockPath
 
       String debug = (String) zs.getProperty(tid, "debug");
 
-      List<String> hlocks = heldLocks.remove(tid);
-      if (hlocks == null)
+      List<String> hlocks = null;
+
+      if (heldLocks == null) {
```

Comment:
If this is made a private internal method then maybe we could assume its never passed null.